bee-network/bee-autopeering/src/config.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Autopeering configuration.
+//!
+//! ## JSON Example
+//! ```json
+//! "autopeering": {
+//!     "enabled": true,
+//!     "bindAddress": "0.0.0.0:14626",
+//!     "entryNodes": [
+//!          "/dns/lucamoser.ch/udp/14826/autopeering/4H6WV54tB29u8xCcEaMGQMn37LFvM1ynNpp27TTXaqNM",
+//!          "/dns/entry-hornet-0.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaPHdAn7eueBnXtikZMwhfPXaeGJGXDt4RBuLuGgb",
+//!          "/dns/entry-hornet-1.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaJJqMd5CQvv1A61coSQCYW9PNT1QKPs7xh2Qg5K2",
+//!          "/dns/entry-mainnet.tanglebay.com/udp/14626/autopeering/iot4By1FD4pFLrGJ6AAe7YEeSu9RbW9xnPUmxMdQenC"
+//!     ],
+//!     "entryNodesPreferIPv6": true,
+//! }
+//! ```
+//!
+//! ## TOML Example
+//!
+//! ```toml
+//! [autopeering]
+//! enabled = true
+//! bind_address = "0.0.0.0:14626"
+//! entry_nodes = [
+//!     "/dns/lucamoser.ch/udp/14826/autopeering/4H6WV54tB29u8xCcEaMGQMn37LFvM1ynNpp27TTXaqNM",
+//!     "/dns/entry-hornet-0.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaPHdAn7eueBnXtikZMwhfPXaeGJGXDt4RBuLuGgb",
+//!     "/dns/entry-hornet-1.h.chrysalis-mainnet.iotaledger.net/udp/14626/autopeering/iotaJJqMd5CQvv1A61coSQCYW9PNT1QKPs7xh2Qg5K2",
+//!     "/dns/entry-mainnet.tanglebay.com/udp/14626/autopeering/iot4By1FD4pFLrGJ6AAe7YEeSu9RbW9xnPUmxMdQenC"
+//! ]
+//! entry_nodes_prefer_ipv6 = true
+//! ```
+
+use crate::multiaddr::AutopeeringMultiaddr;
+
+use serde::{Deserialize, Serialize};
+
+use std::{
+    fmt::Debug,
+    net::{IpAddr, Ipv4Addr, SocketAddr},
+};
+
+// TODO: watch out for possible constification regarding `SocketAddr::new()`.
+const AUTOPEERING_BIND_ADDR_DEFAULT: IpAddr = IpAddr::V4(Ipv4Addr::UNSPECIFIED);
+const AUTOPEERING_BIND_PORT_DEFAULT: u16 = 0;
+const ENTRYNODES_PREFER_IPV6_DEFAULT: bool = false;
+const RUN_AS_ENTRYNODE_DEFAULT: bool = false;
+const DROP_NEIGHBORS_ON_SALT_UPDATE_DEFAULT: bool = false;
+
+/// The autopeering config.
+#[derive(Clone, Debug)]
+pub struct AutopeeringConfig {
+    enabled: bool,
+    bind_addr: SocketAddr,
+    entry_nodes: Vec<AutopeeringMultiaddr>,
+    entry_nodes_prefer_ipv6: bool,
+    run_as_entry_node: bool,
+    drop_neighbors_on_salt_update: bool,
+}
+
+impl AutopeeringConfig {
+    /// Wether autopeering should be enabled.
```

Comment:
```suggestion
    /// Whether autopeering should be enabled.
```

Reply:
I wanted to pay you a lot of respect for catching those typos, but then I saw that you missed the one in line 190..so shame on you!!!

Reply:
:trollface: